File: software/examples/csharp/ExampleThreshold.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Analog In Bricklet 2.0

	// Callback function for voltage reached callback (parameter has unit mV)
	static void VoltageReachedCB(BrickletAnalogInV2 sender, int voltage)
	{
		Console.WriteLine("Voltage: " + voltage/1000.0 + " V");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletAnalogInV2 ai = new BrickletAnalogInV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		ai.SetDebouncePeriod(10000);

		// Register voltage reached callback to function VoltageReachedCB
		ai.VoltageReached += VoltageReachedCB;

		// Configure threshold for voltage "smaller than 5 V" (unit is mV)
		ai.SetVoltageCallbackThreshold('<', 5*1000, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/php/ExampleThreshold.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletAnalogInV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletAnalogInV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Analog In Bricklet 2.0

// Callback function for voltage reached callback
function cb_voltageReached($voltage)
{
    echo "Voltage: " . $voltage/1000.0 . " V\n";
}

$ipcon = new IPConnection(); // Create IP connection
$ai = new BrickletAnalogInV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
$ai->setDebouncePeriod(10000);

// Register voltage reached callback to function cb_voltageReached
$ai->registerCallback(BrickletAnalogInV2::CALLBACK_VOLTAGE_REACHED, 'cb_voltageReached');

// Configure threshold for voltage "smaller than 5 V"
$ai->setVoltageCallbackThreshold('<', 5*1000, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
